app/main.py

```diff
@@ -1,3 +1,56 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int):
+        self.km = km
+
+    def __add__(self, other: int):
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other):
+        if isinstance(other, (int, float)):
+            self.km += other
+        else:
+            self.km += other.km
+        return self
+
+    def __mul__(self, other):
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+
+    def __truediv__(self, other):
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other):
+        if isinstance(other, int):
+            return self.km < other
+        return self.km < other.km
+
+    def __gt__(self, other):
+        if isinstance(other, (int, float)):
+            return self.km > other
+        return self.km > other.km
+
+    def __eq__(self, other):
+        if isinstance(other, int):
+            return self.km == other
+        return self.km == other.km
+
+    def __le__(self, other):
+        if isinstance(other, int):
+            return self.km <= other
+        return self.km <= other.km
```

Comment:
here you can use magic method you already wrote (don't forget to use it as a magic method)
le == not gt